### PR TITLE
feat(rpc): extend search path, remove cd

### PR DESCRIPTION
**Pull request by pnwamk**
Disable the ability for clients to change the current working directory of a running server - instead make it trivial to extend the search path dynamically. See https://github.com/GaloisInc/cryptol/issues/1081

- [x] add a test
- [x] verify the eval server is okay with this feature (it did not have the ability to change a working directory before... but extending the search path seems harmless and unlikely to violate eval server restrictions)
- [ ] investigate if `saw-remote-api` also needs this feature

### CODE REVIEW

**Comment by pnwamk** [comment]
@atomb In this PR I've added the ability for clients to extend the `CRYPTOLPATH` the server uses to find modules for both `cryptol-remote-api` and `cryptol-eval-server`. I just want to double check this functionality doesn't conflict with the goals of the latter.

File: cryptol-remote-api/python/tests/cryptol_eval/test-files/test-subdir/Bar.cry [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module Bar where
```
Comment (atomb) [inline cryptol-remote-api/python/tests/cryptol_eval/test-files/test-subdir/Bar.cry:1]: It's a little unfortunate to need two copies of this file, but it's not really a big deal.

**Review by atomb — APPROVED** [review]
This looks good to me.

File: cryptol-remote-api/python/tests/cryptol_eval/test-files/test-subdir/Bar.cry (hunk 1 above)
Comment (pnwamk) [reply cryptol-remote-api/python/tests/cryptol_eval/test-files/test-subdir/Bar.cry:1]: Yah, the way we have testing set up currently means it's quite convenient to have distinct directories for the standard and eval server... perhaps theres a set of "common test cases" that should work the same for both and we could reduce some duplication that way?

**Pull request merged**